### PR TITLE
Copy capabilities from menu items to wp_navigation

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/54487

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by hellofromtonya — APPROVED** [review]

**Comment by getdave** [comment]
@spacedmonkey @hellofromtonya I think we should adjust this change set to match https://github.com/WordPress/gutenberg/pull/37454/files#diff-68aa3e914d52bce59d8cfcac6c7a10cad3bb8dca6021d00be35bdf546a6ea161R53.

```php
'capabilities'          => array(
    'edit_others_posts'      => 'edit_theme_options',
    'delete_posts'           => 'edit_theme_options',
    'publish_posts'          => 'edit_theme_options',
    'create_posts'           => 'edit_theme_options',
    'read_private_posts'     => 'edit_theme_options',
    'delete_private_posts'   => 'edit_theme_options',
    'delete_published_posts' => 'edit_theme_options',
    'delete_others_posts'    => 'edit_theme_options',
    'edit_private_posts'     => 'edit_theme_options',
    'edit_published_posts'   => 'edit_theme_options',
),
```

If we don't then as things stand in Gutenberg folks with lower permissions won't be able to "read" Navigation posts. That's why we adjusted the permissions as per https://github.com/WordPress/gutenberg/pull/37454#discussion_r771026149.

File: src/wp-includes/post.php [hunk 1]
```diff
@@ -509,6 +509,24 @@ function create_initial_post_types() {
...
+				'delete_posts'           => 'edit_theme_options',
+				'publish_posts'          => 'edit_theme_options',
+				'read_private_posts'     => 'edit_theme_options',
+				'read'                   => 'read',
+				'delete_private_posts'   => 'edit_theme_options',
+				'delete_published_posts' => 'edit_theme_options',
+				'delete_others_posts'    => 'edit_theme_options',
+				'edit_private_posts'     => 'edit_theme_options',
+				'edit_published_posts'   => 'edit_theme_options',
+			),
```
Comment (hellofromtonya) [inline src/wp-includes/post.php]: Matches capabilities of `wp_navigation` CPT in the new merged [GB PR 37454](https://github.com/WordPress/gutenberg/pull/37454/files#diff-68aa3e914d52bce59d8cfcac6c7a10cad3bb8dca6021d00be35bdf546a6ea161R53):

```suggestion
			'capabilities'          => array(
				'edit_others_posts'      => 'edit_theme_options',
				'delete_posts'           => 'edit_theme_options',
				'publish_posts'          => 'edit_theme_options',
				'create_posts'           => 'edit_theme_options',
				'read_private_posts'     => 'edit_theme_options',
				'delete_private_posts'   => 'edit_theme_options',
				'delete_published_posts' => 'edit_theme_options',
				'delete_others_posts'    => 'edit_theme_options',
				'edit_private_posts'     => 'edit_theme_options',
				'edit_published_posts'   => 'edit_theme_options',
			),
```

**Review by audrasjb — APPROVED** [review]
The proposed patch looks good to me 👍

**Comment by getdave** [comment]
@hellofromtonya can we close this one out now?

**Comment by hellofromtonya** [comment]
Thanks for the ping @getdave. Yes, this can be closed.

Committed via https://core.trac.wordpress.org/changeset/52400.

**Pull request closed without merging**